src/spikeinterface/core/core_tools.py

```diff
@@ -934,3 +941,41 @@ def convert_bytes_to_str(byte_value: int) -> str:
         byte_value /= 1024
         i += 1
     return f"{byte_value:.2f} {suffixes[i]}"
+
+
+def spike_vector_to_dict(spike_vector: np.ndarray) -> dict:
+    assert HAVE_NUMBA, "spike_vector_to_dict() requires `numba`!"
+
+    spike_trains = _vector_to_dict(
+        spike_vector["sample_index"].astype(np.int64),
+        spike_vector["unit_index"].astype(np.int64),
+        spike_vector["segment_index"].astype(np.int64),
+    )
+
+    return spike_trains
+
+
+if HAVE_NUMBA:
+
+    @numba.jit((numba.int64[::1], numba.int64[::1], numba.int64[::1]), nopython=True, nogil=True, cache=True)
+    def _vector_to_dict(sample_index, unit_index, segment_index):
+        spike_trains = []
+        n_units = 1 + np.max(unit_index)
+
+        for seg in range(1 + segment_index[-1]):
+            n_spikes = np.zeros(n_units, dtype=np.int64)
+            for i in range(len(sample_index)):
+                n_spikes[unit_index[i]] += 1
+
+            d = numba.typed.Dict()
```

Comment:
Let's avoid single variable names that crash with the debugger. More explicit is better but if you want to use a single letter variable use something other than d.